test/tests/restart/restart_transient_from_eigen/tests

```diff
@@ -0,0 +1,19 @@
+[Tests]
+  design = "Restartable.md"
+  issues = '#21011'
+
+  [eigen_store_solution]
+    type = 'Exodiff'
+    input = 'eigen.i'
+    exodiff = 'eigen_out.e'
+    requirement = 'The system shall support storing the solution of an eigenvalue solve such that it can be restarted later into a non-eigenvalue solve'
```

Comment:
```suggestion
    requirement = 'The system shall support storing the solution of an eigenvalue solve such that it can be restarted later into a non-eigenvalue solve.'
```